src/grid/storage.rs

```diff
@@ -200,8 +197,19 @@ impl<T> Storage<T> {
     }
 
     /// Compute actual index in underlying storage given the requested index.
-    fn compute_index(&self, requested: usize) -> usize {
-        (requested + self.zero) % self.inner.len()
+    fn compute_index(&self, mut requested: usize) -> usize {
+        // If line outside the buffer is requested, return the last line in the buffer
```

Comment:
Could we just `assert!` on this condition rather than using `min`? This would force us to improve bookkeeping elsewhere, but it would potentially allow this method to be further optimized.

Reply:
Yeah I've thought about that. My intention for now was to keep things the way they are, however this could definitely cause issues if anyone else is trying to request a size which is too big (at least this might cause silent failure, or failure at the wrong location).

I'm all for changing this though, however the performance part of it shouldn't be an issue. Currently this is not a problem at all and It's not a super slow way to achieve this either, so I wouldn't do it for the performance but rather because it feels a bit more like the "proper" way to do things.

Which just leaves the question if selection should be completely cancelled or if selection should go up to the last line in the buffer. I'm all for keeping the behavior it is right now and making it go to the last line of the buffer. I don't think there's any issue with selection potentially returning an "incorrect" result, so something's better than nothing I'd say.

Reply:
Done! Seems to work out well. Decided to go for a `debug_assert` just to make sure there's no unnecessary slowdowns in release mode.